main/src/cgeo/geocaching/connector/gc/GCMap.java

```diff
@@ -134,19 +136,22 @@ public static SearchResult searchByViewport(@NonNull final Viewport viewport) {
     }
 
     @NonNull
-    public static SearchResult searchByFilter(@NonNull final GeocacheFilter filter) {
+    public static SearchResult searchByFilter(@NonNull final GeocacheFilter filter, final IConnector connector) {
         final GCWebAPI.WebApiSearch search = new GCWebAPI.WebApiSearch();
         search.setOrigin(Sensors.getInstance().currentGeo().getCoords());
         search.setPage(200, 0);
 
         for (BaseGeocacheFilter baseFilter: filter.getAndChainIfPossible()) {
+            if (baseFilter instanceof OriginGeocacheFilter && !((OriginGeocacheFilter) baseFilter).allowsCachesOf(connector)) {
+                return new SearchResult(); //no need to search if connector is filtered out itself
+            }
             fillForBasicFilter(baseFilter, search);
         }
 
         return GCWebAPI.searchCaches(search);
     }
 
-    private static void fillForBasicFilter(@NonNull final BaseGeocacheFilter basicFilter, final GCWebAPI.WebApiSearch search) {
+    private static boolean fillForBasicFilter(@NonNull final BaseGeocacheFilter basicFilter, final GCWebAPI.WebApiSearch search) {
```

Comment:
Really change this to return a value?
The used code insn't changed, the return is not used.

Reply:
You're right. An artefact of my first implementation solution, this should have been reverted by me.

Since PR is already merged, I will fix this with another one.

Reply:
Fixed in PR #10920